### PR TITLE
Ease IdeaModel configuration through Kotlin extensions

### PR DESCRIPTION
**Pull request by jprinet**
Goal is to be able to use idiomatic configuration blocks in `build.gradle.kts` like below:

```
idea {
    settings {
        compiler {
            processHeapSize = 4096
            parallelCompilation = false
            javac {
                javacAdditionalOptions = listOf(
                    "-Werror",
                    "-Xlint:all",
                    "-Xlint:-options",
                    "-Xlint:-serial",
                    "-proc:none",
                    "-Xdiags:verbose",
                    "-parameters"
                ).joinToString(separator = " ")
            }
        }
    }
}
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -50,3 +54,7 @@ wrapper {
   gradleVersion = "5.6.4"
   distributionType = Wrapper.DistributionType.BIN
 }
+
+compileGroovy.dependsOn.remove('compileJava')
```
Comment (runningcode) [inline build.gradle:58]: Why is this required?

File: build.gradle [hunk 2]
```diff
@@ -50,3 +54,7 @@ wrapper {
...
+compileKotlin.dependsOn compileGroovy
```
Comment (runningcode) [inline build.gradle:59]: I wonder if this is fixed by  using a more recent version of KGP? 🤔

File: build.gradle [hunk 3]
```diff
@@ -15,6 +17,8 @@ repositories {
 dependencies {
   implementation 'com.google.code.gson:gson:2.8.6'
   implementation 'com.google.guava:guava:28.2-jre'
+  implementation ProjectExtensionsKt.gradleKotlinDsl(project)
```
Comment (runningcode) [inline build.gradle:20]: why is this needed?

File: src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package org.jetbrains.gradle.ext
```
Comment (runningcode) [inline src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt:1]: i was under the impression that these extension functions needed to be in the `org.gradle.kotlin.dsl` package. why did you place them in this package?

File: build.gradle (hunk 1 above)
Comment (jprinet) [reply build.gradle:58]: We need to call Groovy code from Kotlin, see this for more details:
https://github.com/gradle/gradle/issues/11333

File: build.gradle (hunk 2 above)
Comment (jprinet) [reply build.gradle:59]: Gradle 6.1 gives a more regular way (see shared thread above)

File: build.gradle (hunk 3 above)
Comment (jprinet) [reply build.gradle:20]: applying `kotlin-dsl` on a _Groovy_ script leads to other issues, here I just need some dependencies provided by the `kotlin-dsl` jar, this approach was suggested by @eskatos

File: src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt (hunk 4 above)
Comment (jprinet) [reply src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt:1]: This is just the unique package that gets exposed from within this plugin, I'm opened to any suggestion though

File: build.gradle (hunk 1 above)
Comment (runningcode) [reply build.gradle:58]: Ah thanks for the pointer. 👍

File: build.gradle (hunk 2 above)
Comment (runningcode) [reply build.gradle:59]: Thanks

File: build.gradle (hunk 3 above)
Comment (runningcode) [reply build.gradle:20]: SGTM

File: src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt (hunk 4 above)
Comment (runningcode) [reply src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt:1]: @eskatos can you clarify why this belongs in `org.jetbrains.gradle.ext` and not `org.gradle.kotlin.dsl`?

Comment (eskatos) [reply src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt:1]: Because this is this plugin's package. Putting plugin code in `org.gradle.kotlin.dsl` in order for them to be available without import is an abusive workaround for the lack of https://github.com/gradle/gradle/issues/7557.

Comment (runningcode) [reply src/main/kotlin/org/jetbrains/gradle/ext/IdeaModelUtils.kt:1]: Ah, I didn't know that. Thanks @eskatos !

**Review by runningcode — APPROVED** [review]

**Comment by nskvortsov** [comment]
Thank you so much for the contribution! Still, some more kotlin-dsl constructs are required to cover all available configurations. Is it correct?

**Comment by jprinet** [comment]
> Thank you so much for the contribution! Still, some more kotlin-dsl constructs are required to cover all available configurations. Is it correct?

Well, it should be enough for the current need
